src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Automation/Peers/WindowAutomationPeer.cs

```diff
@@ -88,6 +88,12 @@ override protected Rect GetBoundingRectangleCore()
 
             return bounds;
         }
+
+        protected override bool IsDialogCore()
+        {
+            Window window = (Window)Owner;
+            return window.IsShowingAsDialog;
```

Comment:
If the app has set the window's IsDialog property, we should return that.  This lets apps override our behavior.  I can't think of a reason they'd want to, but that doesn't mean there isn't one.

```suggestion
if (MS.Internal.Helper.IsDefaultValue(AutomationProperties.IsDialogProperty, window)
{
  return window.IsShowingAsDialog;
}
else
{
   return AutomationProperties.GetIsDialog(window);
}
```